plugins/crowdstrike/custom_actions_test.go

```diff
@@ -0,0 +1,254 @@
+
+	// create a context with the connection
+	connection := map[string]*connections.ConnectionInstance{
+		"crowdstrike": {
+			Data: map[string]string{"REQUEST_URL": suite.testServer.URL, "Client ID": "mock_crowdstrike_client_id", "Client Secret": "mock_crowdstrike_client_secret"},
+		},
+	}
+
+	suite.context = plugin.NewActionContext(nil, connection)
+}
+
+func GetGinRouter() *gin.Engine {
+	router := gin.New()
+
+	AddHandleAuthEndpoint(router)
+	AddDeleteDeviceEndpoint(router)
+	AddGetDevicesEndpoint(router)
+	AddGetDevicesWithInfo(router)
+
+	return router
+}
+
+func AddHandleAuthEndpoint(router *gin.Engine) {
+	router.POST("/oauth2/token", func(c *gin.Context) {
+		c.JSON(http.StatusCreated, gin.H{"access_token": "mock_crowdstrike_access_token"})
+	})
+}
+
+func AddDeleteDeviceEndpoint(router *gin.Engine) {
+	router.POST("/devices/entities/devices-actions/v2", func(c *gin.Context) {
+		actionName := c.Request.URL.Query().Get("action_name")
+		if actionName != "hide_host" {
+			c.JSON(http.StatusNotFound, gin.H{"success": false, "error": "page not found"})
+		}
+
+		var requestBody DeleteDeviceRequest
+		if err := c.BindJSON(&requestBody); err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"success": false, "error": "invalid request body"})
+		}
+
+		if len(requestBody.DeviceID) == 1 && requestBody.DeviceID[0] == device1 {
+			c.JSON(http.StatusOK, gin.H{"success": true})
+		} else {
+			c.JSON(http.StatusNotFound, gin.H{"success": false, "error": "device not found"})
+		}
+	})
+}
+
+func AddGetDevicesEndpoint(router *gin.Engine) {
+	router.GET("/devices/queries/devices/v1", func(c *gin.Context) {
+		c.JSON(http.StatusOK, gin.H{
+			"meta": map[string]interface{}{
+				"pagination": map[string]int{
+					"total": 2,
+				},
+			},
+			"resources": []string{
+				device1,
+				device2,
+			},
+		})
+	})
+}
+
+func AddGetDevicesWithInfo(router *gin.Engine) {
+	router.GET("/devices/entities/devices/v1", func(c *gin.Context) {
+		ids, ok := c.Request.URL.Query()["ids"]
+
+		if !ok || len(ids) <= 0 || ids[0] != device1 {
+			// device doesn't exist
+			c.JSON(http.StatusNotFound, gin.H{"errors": map[string]interface{}{"code": 404, "message": "No matching device found for ID"}})
+		} else if len(ids) == 1 && ids[0] == device1 {
+			// one device
+			c.JSON(http.StatusOK, gin.H{
+				"resources": []map[string]interface{}{
+					{
+						"device_id":                    device1,
+						"serial_number":                serial1,
+						"detection_suppression_status": "suppressed",
+					},
+				},
+			})
+		} else if len(ids) == 2 && ids[0] == device1 && ids[1] == device2 {
+			// two devices
+			c.JSON(http.StatusOK, gin.H{
+				"resources": []map[string]interface{}{
+					{
+						"device_id":                    device1,
+						"serial_number":                serial1,
+						"detection_suppression_status": "suppressed",
+					},
+					{
+						"device_id":     device2,
+						"serial_number": serial2,
+					},
+				},
+			})
+		}
```

Comment:
There's no response if none of the if statements hit
Would be better to do
```suggestion
  if x
    return 
  if y
    return
  if z
    return
  return
```